src/pybind/mgr/dashboard/frontend/src/app/shared/models/osd.model.ts

```diff
@@ -0,0 +1,43 @@
+export interface Osd {
+  id?: number;
```

Comment:
why do we have all field optional ?

Reply:
i made some of the optional and some of them required now.

Reply:
Okay, and what is the deciding criteria ? Or if we can have some comment here for future, will help, thanks!

Reply:
just added a comment there in the file. But there is not a definite schema in the back-end so its pretty difficult to understand what are all we need in the UI. To avoid any regressive behavior I just went with what I saw in some of the test cases as the deciding factor for making some items optional and non-optional.

Reply:
We should def type things at the back-end and use some tooling to dynamically generate front-end models from that (Python typing -> JSONSchema -> Typescript model).